src/commands/run.rs

```diff
@@ -376,13 +391,54 @@ impl RunCommand {
         // Invoke the function and then afterwards print all the results that came
         // out, if there are any.
         let mut results = vec![Val::null(); ty.results().len()];
-        func.call(store, &values, &mut results).with_context(|| {
+        let invoke_res = func.call(store, &values, &mut results).with_context(|| {
             if let Some(name) = name {
                 format!("failed to invoke `{}`", name)
             } else {
                 format!("failed to invoke command default")
             }
-        })?;
+        });
+        match invoke_res {
+            Ok(_) => {}
+            Err(err) => {
+                if let Some(coredump_path) = self.coredump_on_trap.as_ref() {
+                    let bt = err
+                        .downcast_ref::<WasmBacktrace>()
+                        .ok_or_else(|| anyhow!("failed to downcast to WasmBacktrace"))?;
+
+                    let mut coredump_builder = wasm_coredump_builder::CoredumpBuilder::new()
+                        .executable_name(&self.module.to_str().unwrap_or_else(|| "unknown"));
+
+                    {
+                        let mut thread_builder =
+                            wasm_coredump_builder::ThreadBuilder::new().thread_name("main");
+
+                        for frame in bt.frames() {
+                            let coredump_frame = wasm_coredump_builder::FrameBuilder::new()
+                                .funcidx(frame.func_index())
+                                .build();
```

Comment:
The backtrace frames here should also have offset information to get to the precise opcode if you'd like to add that.

Reply:
Yeah, we'll probably want that in the future. I haven't changed (or decided) the coredump format to allow `funcidx` in addition to `codeoffset` yet.